### ISSUE TITLE
Hamster does not start

### ISSUE THREAD

**Issue by reinhardt** [NixOS/nixpkgs#27498]
## Issue description

When trying to start hamster (package hamster-time-tracker) it fails with

```
WARNING:root:Could not import wnck - workspace tracking will be disabled
Traceback (most recent call last):
  File "/nix/store/vzknypbn57cf9bc03wrgsfpfqrka3rgj-hamster-time-tracker-1.04/bin/.hamster-wrapped", line 375, in <module>
    hamster_client.today()
  File "/nix/store/vzknypbn57cf9bc03wrgsfpfqrka3rgj-hamster-time-tracker-1.04/bin/.hamster-wrapped", line 137, in today
    app = today.DailyView()
  File "/nix/store/vzknypbn57cf9bc03wrgsfpfqrka3rgj-hamster-time-tracker-1.04/lib/python2.7/site-packages/hamster/today.py", line 90, in __init__
    self.create_hamster_window()
  File "/nix/store/vzknypbn57cf9bc03wrgsfpfqrka3rgj-hamster-time-tracker-1.04/lib/python2.7/site-packages/hamster/today.py", line 148, in create_hamster_window
    self.new_name = widgets.ActivityEntry()
  File "/nix/store/vzknypbn57cf9bc03wrgsfpfqrka3rgj-hamster-time-tracker-1.04/lib/python2.7/site-packages/hamster/widgets/activityentry.py", line 103, in __init__
    self.populate_suggestions()
  File "/nix/store/vzknypbn57cf9bc03wrgsfpfqrka3rgj-hamster-time-tracker-1.04/lib/python2.7/site-packages/hamster/widgets/activityentry.py", line 234, in populate_suggestions
    self.activities = runtime.storage.get_activities(fact.activity)
  File "/nix/store/vzknypbn57cf9bc03wrgsfpfqrka3rgj-hamster-time-tracker-1.04/lib/python2.7/site-packages/hamster/client.py", line 137, in get_activities
    return self._to_dict(('name', 'category'), self.conn.GetActivities(search))
  File "/nix/store/vzknypbn57cf9bc03wrgsfpfqrka3rgj-hamster-time-tracker-1.04/lib/python2.7/site-packages/hamster/client.py", line 88, in conn
    '/org/gnome/Hamster'),
  File "/nix/store/y1g1k0qmdyx4pq4i7pi70kwg077d4mld-python2.7-dbus-python-1.2.4/lib/python2.7/site-packages/dbus/bus.py", line 241, in get_object
    follow_name_owner_changes=follow_name_owner_changes)
  File "/nix/store/y1g1k0qmdyx4pq4i7pi70kwg077d4mld-python2.7-dbus-python-1.2.4/lib/python2.7/site-packages/dbus/proxies.py", line 248, in __init__
    self._named_service = conn.activate_name_owner(bus_name)
  File "/nix/store/y1g1k0qmdyx4pq4i7pi70kwg077d4mld-python2.7-dbus-python-1.2.4/lib/python2.7/site-packages/dbus/bus.py", line 180, in activate_name_owner
    self.start_service_by_name(bus_name)
  File "/nix/store/y1g1k0qmdyx4pq4i7pi70kwg077d4mld-python2.7-dbus-python-1.2.4/lib/python2.7/site-packages/dbus/bus.py", line 278, in start_service_by_name
    'su', (bus_name, flags)))
  File "/nix/store/y1g1k0qmdyx4pq4i7pi70kwg077d4mld-python2.7-dbus-python-1.2.4/lib/python2.7/site-packages/dbus/connection.py", line 651, in call_blocking
    message, timeout)
dbus.exceptions.DBusException: org.freedesktop.DBus.Error.Spawn.ChildExited: Process org.gnome.Hamster exited with status 1
```

Manually starting lib/hamster-time-tracker/hamster-service gives

```
Traceback (most recent call last):
  File "/nix/store/glipkbng5k3nxxqhb7gh25ddi6sknwsn-hamster-time-tracker-1.04/lib/hamster-time-tracker/hamster-service", line 4, in <module>
    import gobject, dbus, dbus.service
ImportError: No module named gobject
```

This call, however, works for me (inside a nix-shell -p hamster-time-tracker):

```
PYTHONPATH=/nix/store/6ndwyx7lb3p9i7n01x98xb7yjhwkb58c-python2.7-pygobject-2.28.6/lib/python2.7/site-packages/:$PYTHONPATH /nix/store/glipkbng5k3nxxqhb7gh25ddi6sknwsn-hamster-time-tracker-1.04/lib/hamster-time-tracker/hamster-service
```

After that hamster runs fine.

Now I'm not sure how to modify the package expression so that the extra PYTHONPATH is not necessary any more. Could someone give me a pointer please?

## Technical details

* System: 17.09.git.1e1472e (Hummingbird)
* Nix version: nix-env (Nix) 1.11.11
* Nixpkgs version: 17.09.git.1e1472ed93M
* Sandboxing enabled: build-use-sandbox = false

_EDIT:_ Clarified that the modified call happens inside a nix-shell

**Comment by FRidh** [comment]
Do you need to be running Gnome or so?
```
$ hamster 
Traceback (most recent call last):
  File "/nix/store/y3ys0j606xi2kafdd9cpaybkkn14vbvy-hamster-time-tracker-1.04/bin/.hamster-wrapped", line 31, in <module>
    from hamster import client, reports
  File "/nix/store/y3ys0j606xi2kafdd9cpaybkkn14vbvy-hamster-time-tracker-1.04/lib/python2.7/site-packages/hamster/reports.py", line 31, in <module>
    from configuration import runtime
  File "/nix/store/y3ys0j606xi2kafdd9cpaybkkn14vbvy-hamster-time-tracker-1.04/lib/python2.7/site-packages/hamster/configuration.py", line 304, in <module>
    conf = GConfStore()
  File "/nix/store/y3ys0j606xi2kafdd9cpaybkkn14vbvy-hamster-time-tracker-1.04/lib/python2.7/site-packages/hamster/configuration.py", line 188, in __init__
    self._client.add_dir(self.GCONF_DIR[:-1], gconf.CLIENT_PRELOAD_RECURSIVE)
glib.GError: Failed to contact configuration server; the most common cause is a missing or misconfigured D-Bus session bus daemon. See http://projects.gnome.org/gconf/ for information. (Details -  1: GetIOR failed: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name org.gnome.GConf was not provided by any .service files)
```

Anyway, I checked the wrapper and it does include `pygobject`. Maybe the problem is that it calls itself but accidentally calls the unwrapped executable. That happens more often.

**Comment by FRidh** [comment]
cc maintainer @bjornfor

**Comment by bjornfor** [comment]
I tested it now an got missing dbus service (like in the post above). It can be fixed with this config:

```
  services.dbus.packages = with pkgs; [ gnome3.gconf hamster-time-tracker ];
```

**Comment by reinhardt** [comment]
This seems to be heading in a slightly different direction now. I don't get the missing dbus service error, only the ChildExited caused by the ImportError. Are we discussing two separate problems?

**Comment by reinhardt** [comment]
I have `gnome2.GConf` in my `dbus.packages`. Changing it to `gnome3.gconf` and/or adding `hamster-time-tracker` does not solve the problem.

I noticed that the `PYTHONPATH=...` is actually unnecessary. As long as I run hamster-service in the nix-shell it works. Outside it fails.

I also noticed that `bin/.hamster-wrapped` has a line that calls `site.addsitedir()` for a bunch of paths, including the pygobject and dbus-python site-packages directories. This line does not seem to come from upstream (see https://github.com/projecthamster/hamster/blob/master/src/hamster-cli) but from wrapPython. `lib/hamster-time-tracker/hamster-service` does not have this line. Could that be the problem, i.e. does wrapPython need to be run on the scripts in the lib dir as well as the bin dir?
